### PR TITLE
Add build/script for nonlib pkg

### PR DESCRIPTION
**Pull request by mgreminger**
This pull request follows from the discussion around #1034. It's an extension to #1027, which adds `build/script` to `meta.yaml` for library packages. This pull request adds `build/script` as an optional section for Python packages. The script is run before `setup.py` is run by `buildpkg.py`. I've also added documentation for the `build/script` and `build/library` options to the `new_packages.md` file.

### CODE REVIEW

File: pyodide_build/buildpkg.py [hunk 1]
```diff
@@ -226,8 +226,10 @@ def run_script(buildpath: Path, srcpath: Path, pkg: Dict[str, Any]):
     finally:
         os.chdir(orig_path)
 
-    with open(buildpath / ".packaged", "wb") as fd:
-        fd.write(b"\n")
+    # If library, we're done so create .packaged file
+    if pkg["build"].get("library"):
+        with open(buildpath / ".packaged", "wb") as fd:
+            fd.write(b"\n")
```
Comment (rth) [inline pyodide_build/buildpkg.py:232]: Would you mind adding a short test to, `pyodide_build/tests/test_buildpkg.py`? With something like,
```py
from textwrap import dedent 

@pytest.mark.parametrize('is_library', [True, False])
def test_run_script(is_library, tmpdir):
    build_dir = tmpdir.mkdir('build')
    src_dir = tmpdir.mkdir('build')
    script = dedent('''
         touch out.txt
    """)
    pkg = {'build': {'script': script}}
    run_script(build_dir, src_dir, pkg)
    assert (build_dir / 'out.txt').exists()
    if library:
        assert (build_dir / '.packaged').exists()
    else:
        assert not (build_dir / '.packaged').exists()
```
(Paths and some details might need adjusting, I haven't tested it). I realize we should have done this when initially adding this feature, but now is also good.

**Review by rth — COMMENTED** [review]
Thanks for adding it and the documentation @mgreminger !

**Comment by mgreminger** [comment]
Makes sense, I'll update the PR to include the test.

File: pyodide_build/tests/test_buildpkg.py [hunk 2]
```diff
@@ -33,3 +36,21 @@ def test_download_and_extract(monkeypatch):
...
+
+@pytest.mark.parametrize("is_library", [True, False])
+def test_run_script(is_library, tmpdir):
+    build_dir = Path(tmpdir.mkdir("build"))
+    src_dir = Path(tmpdir.mkdir("build/package_name"))
+    script = dedent(
+        """
+         touch out.txt
+    """
+    )
```
Comment (dalcde) [inline pyodide_build/tests/test_buildpkg.py:49]:
```suggestion
    script = "touch out.txt"
```

File: pyodide_build/tests/test_buildpkg.py [hunk 3]
```diff
@@ -1,8 +1,11 @@
 import shutil
 import subprocess
+from textwrap import dedent
```
Comment (rth) [inline pyodide_build/tests/test_buildpkg.py:3]:
```suggestion
import subprocess
```

**Comment by rth** [comment]
Thanks! I applied the review suggestion and removed an import that's no longer necessary. Will merge once the CI passes.

**Pull request merged**